openff/toolkit/_tests/test_forcefield.py

```diff
@@ -1711,6 +1711,23 @@ def test_lookup_parameter_type(self, force_field):
         ):
             force_field["vdW"][smirks]
 
+    def test_lookup_duplicate_parameter_type(self, force_field):
+        """Test that if a parameter SMIRKS appears twice, lookups will yield the second copy"""
+        first_bond = force_field["Bonds"][0]
+        force_field["Bonds"].add_parameter(
+            {
+                "smirks": first_bond.smirks,
+                "k": first_bond.k / 2,
+                "length": first_bond.length / 2,
+                "id": first_bond.id + "2",
+            },
+            allow_duplicate_smirks=True,
+        )
+        result = force_field["Bonds"][first_bond.smirks]
```

Comment:
Worth testing at both `ParameterHandler` and `ParameterList`? i.e.
```suggestion
        result = force_field["Bonds"][first_bond.smirks]
        assert result == forcefield['Bonds'].parameters[first_bond.smirks]
```

Reply:
Good call, thanks!